docs/user_guide.md

```diff
@@ -1008,10 +1117,47 @@ If you set a quota on a project, this quota cannot be exceeded. The quota is app
 
 **WARNING**: You cannot revert a rule after you run it. It is strongly recommended to perform a dry run before you run rules. 
 
-To modify an existing rule, click the three vertical dots next to a rule to disable, edit, or delete that rule. 
+To modify an existing rule, use the **Action** drop-down menu next to a rule to disable, edit, or delete that rule. 
 
 ![Modify tag retention rules](img/tag-retention5.png)
 
+## Configure Tag Immutability
+
+By default, users can repeatedly push an image with the same tag to repositories in Harbor. This causes the previous image to effectively be overwritten with each push, in that the tag now points to a different image and the image that previously used the tag now becomes tagless. This is due to the Docker implementation, that does not enforce the mapping between an image tag and the image digest. This can be undesirable in certain cases, because the tag can no longer be trusted to identify the image version. The sha256 digest remains reliable and always points to the same build, but it is not rendered in a human-friendly format.  
+
+Moreover, the Docker implementation requires that deleting a tag results in the deletion of all other tags that point to the same digest, causing unwanted image deletions.
+
+To prevent this, Harbor allows you to configure tag immutability at the project level, so that images with certain tags cannot be pushed into Harbor if their tags match existing tags. This prevents existing images from being overwritten. Tag immutability guarantees that an immutable tagged image cannot be deleted, and cannot be altered through repushing, retagging, or replication.
+
+1. Log in to the Harbor interface with an account that has at least project administrator privileges.
+1. Go to **Projects**, select a project, click the **More** ellipsis (`...`), and select **Tag Strategy**.
+
+   ![Tag options](img/tag-retention1.png)
+1. Select **Tag Immutability**.
+
+   ![Add an immutability rule](img/tag-immutability.png)
+1. Click **Add Rule**.
+
+   - In the **For the respositories** row, enter a comma-separated list of repositories to which to either apply or exclude from the rule by selecting either **matching** or **excluding** from the drop-down menu.
+   - In the **Tags** row, enter a comma-separated list of tags to which to either apply or exclude from the rule by selecting either **matching** or **excluding** from the drop-down menu.
+ 
+   ![Add an immutability rule](img/add-immutability-rule.png)
+1. Click **Add** to save the rule.
+
+For example, to make all tags for all repositories in the project immutable, set the following options:
```

Comment:
the algorithm between rules is `OR`, it means that if one tag is matched by any of rules, it's a immutable rule. Do we need to add this?

Reply:
@wy65701436 I added this in the intro in https://github.com/goharbor/harbor/pull/10029/commits/a1eac99a49aa192cb5048c5eaa8a4e2a756cd134.